File: src/test/java/boj/problems/step7/No2292Test.java
package boj.problems.step7;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.StringReader;
import java.io.StringWriter;
import java.util.stream.Collectors;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class No2292Test {
    static class 벌집_테스트케이스 {
        int given;
        int want;

        public 벌집_테스트케이스(int given, int want) {
            this.given = given;
            this.want = want;
        }
    }

    @Test
    @DisplayName("벌집 테스트")
    void 벌집_테스트() throws IOException {
        System.out.println("벌집 : https://www.acmicpc.net/problem/2292");

        벌집_테스트케이스[] testCase = {
                new 벌집_테스트케이스(13, 3),
                new 벌집_테스트케이스(1, 1),
                new 벌집_테스트케이스(2, 2),
                new 벌집_테스트케이스(7, 2),
                new 벌집_테스트케이스(5, 2),
                new 벌집_테스트케이스(37, 4),
                new 벌집_테스트케이스(38, 5),
                new 벌집_테스트케이스(64, 6)
        };

        for (벌집_테스트케이스 tc : testCase) {
            BufferedReader br_given = new BufferedReader(new StringReader(String.valueOf(tc.given)));
            StringWriter sw = new StringWriter();
            BufferedWriter bw_got = new BufferedWriter(sw);

            No2292.solve(br_given, bw_got);

            bw_got.close();

            BufferedReader br_got = new BufferedReader(new StringReader(sw.getBuffer().toString()));

            String got = br_got.lines().collect(Collectors.joining());

            assertEquals(tc.want, Integer.parseInt(got));
        }
    }
}

File: src/test/kotlin/boj/problems/step7/No2292Test.kt
package boj.problems.step7

import io.kotest.core.spec.style.StringSpec
import io.kotest.data.row
import io.kotest.matchers.shouldBe
import java.io.BufferedReader
import java.io.StringReader

class No2292Test : StringSpec() {
    init {
        "벌집 테스트" {
            println("벌집 : https://www.acmicpc.net/problem/2292")

            val testCases = listOf(
                row(13, 3),
                row(1, 1),
                row(2, 2),
                row(7, 2),
                row(5, 2),
                row(37, 4),
                row(38, 5),
                row(64, 6)
            )

            testCases.forEach { (input, expected) ->
                val actual = No2292.solve(BufferedReader(StringReader(input.toString())))
                actual shouldBe expected
            }

        }
    }
}
